crates/aptos/src/move_tool/mod.rs

```diff
@@ -222,6 +226,47 @@ impl CliCommand<&'static str> for TestPackage {
     }
 }
 
+/// Prove the Move package at the package path
+#[derive(Parser)]
+pub struct ProvePackage {
+    #[clap(flatten)]
+    move_options: MovePackageDir,
+
+    /// A filter string to determine which unit tests to run
+    #[clap(long)]
+    pub filter: Option<String>,
+}
+
+#[async_trait]
+impl CliCommand<&'static str> for ProvePackage {
+    fn command_name(&self) -> &'static str {
+        "ProvePackage"
+    }
+
+    async fn execute(self) -> CliTypedResult<&'static str> {
+        let config = BuildConfig {
+            additional_named_addresses: self.move_options.named_addresses(),
+            test_mode: true,
+            install_dir: self.move_options.output_dir.clone(),
+            ..Default::default()
+        };
+        let result = task::spawn_blocking(move || {
```

Comment:
A blocking call is need here. Otherwise, it has an issue with the `async` calls used within `run_move_prover` (more specifically, in `boogie_wrapper`).